Girder & machine frame disassembly changes

kingofkosmos opened the pull request:
Old way:
Wrenching a girder (unsecured) -> secures the girder.
Wrenching a girder (secured) -> disassembles the girder.
Crowbarring a girder (secured) -> unsecures the girder.

**New way:**
Wrenching a girder (unsecured) -> secures the girder (same as before).
Wrenching a girder (secured) -> unsecures the girder.
Welding a girder (unsecured) -> disassembles the girder.

Also:
**Machine frames** can now be secured/unsecured with a wrench and disassembled with a welder. Old way was that they couldn't be unsecured at all, and disassembly was done with a wrench.

This way they're more consistent with nearly every other frame in the game, which have to be first unsecured with a wrench and then disassembled with a welder.

kingofkosmos committed:
Girder disassembly changes

Current situation:
Wrench to girder (unsecured) -> secure girder
Wrench to girder (secured) -> disassemble girder
Crowbar to girder (secured) -> unsecure girder

This PR changes it to:
Wrench to girder (unsecured) -> secure girder (same as before)
Wrench to girder (secured) -> unsecure girder
Welder to girder (unsecured) -> disassemble girder

This is more consistent with nearly every other frame in the game, which have to be first unsecured with a wrench and then disassembled with a welder.

kingofkosmos committed:
changelog addition

kingofkosmos commented:
Also fixes an unreported issue about securing an unsecured girder. You could start securing it, push it away and it would secure itself after a while, while being a tile away from you.

Miauw62 commented:
Makes sense :+1:
Op 4-apr.-2015 20:15 schreef "kingofkosmos" <notifications@github.com>:

> Old way:
> Wrenching a girder (unsecured) -> secures the girder.
> Wrenching a girder (secured) -> disassembles the girder.
> Crowbarring a girder (secured) -> unsecures the girder.
>
> *New way:*
> Wrenching a girder (unsecured) -> secures the girder (same as before).
> Wrenching a girder (secured) -> unsecures the girder.
> Welding a girder (unsecured) -> disassembles the girder.
>
> This is more consistent with nearly every other frame in the game, which
> have to be first unsecured with a wrench and then disassembled with a
> welder.
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/tgstation/-tg-station/pull/8741
> Commit Summary
>
>    - Girder disassembly changes
>    - changelog addition
>
> File Changes
>
>    - *M* code/game/objects/structures/girders.dm
>    <https://github.com/tgstation/-tg-station/pull/8741/files#diff-0>
>    (121)
>    - *A* html/changelogs/kingofkosmos-girderrakennus.yml
>    <https://github.com/tgstation/-tg-station/pull/8741/files#diff-1> (6)
>
> Patch Links:
>
>    - https://github.com/tgstation/-tg-station/pull/8741.patch
>    - https://github.com/tgstation/-tg-station/pull/8741.diff
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/tgstation/-tg-station/pull/8741>.
>

chimersable commented:
Does it work correctly with the plasma cutter, thermite, or other wall mod effects?

phil235 commented on code/game/objects/structures/girders.dm:
```diff
@@ -4,48 +4,55 @@
+				transfer_fingerprints_to(M)
 				qdel(src)
 
+	else if(istype(W, /obj/item/weapon/wrench) && state == 3 && 4)
```
why the `&& 4` ?

kingofkosmos replied:
Good eye! I forgot it there after testing if I could remove 90% of the cultgirder's mostly-duplicatecode and change cultgirder into just a state, but soon realized that this stuff was too advanced for me. I'll remove it.

kingofkosmos committed:
Changed to only user's fingerprints get to the new metal after disassembly.

kingofkosmos commented:
>Does it work correctly with the plasma cutter, thermite, or other wall mod effects?
Plasma cutter works but spawns only 1 metal per girder. I'll make it spawn 2 metal sheets, is that ok? I'm not sure if I can make it spawn a plasteel after slicing a reinforced girder.
Thermite doesn't work on girders. Jackhammering the girder works as normal.
I'm not sure what you mean by wall mod effects. Wall stuff is in a different file, unaffected by this.

kingofkosmos committed:
Removed the extra 4. Plasma cutting now produces 2 metals per girder.

ACCount12 commented:
Keep crowbar unsecuring please, because emergency toolbox has only crowbar and welder in it.

MrPerson commented:
What does that have to do with anything? If you want the emergency toolbox to continue to be able to cut down walls, give it a welder

RemieRichards commented on code/game/objects/structures/girders.dm:
```diff
@@ -4,48 +4,56 @@
-				return
-			playsound(src.loc, 'sound/items/Ratchet.ogg', 100, 1)
-			user << "<span class='notice'>Now securing the girder...</span>"
+	if(istype(W, /obj/item/weapon/weldingtool) && state == 3)
```
replace these ```state == 3``` Checks with ```state == GIRDER_DISPLACED```

At the top of this file you'll want to add
```
#define GIRDER_NORMAL 0
#define GIRDER_REINF_STRUTS 1
#define GIRDER_REINF 2
#define GIRDER_DISPLACED 3
```
And then replace all the other checks with these too.

ACCount12 commented:
@MrPerson, it has a welder since  #8668.

KorPhaeron commented:
So now it takes 3 steps/2 uses of welder fuel to take apart even basic walls?

I mean it adds consistency but it also adds extra steps to a simple action

ACCount12 commented:
@KorPhaeron, yes, that's why i dislike it too.

optimumtact commented:
Who actually fully disassembles walls?

A little extra fuel usage isn't going to hurt in that case because it implies you're doing large scale construction, so you should've brought a welding tank with you

PKPenguin321 commented:
I dislike this. Deconstructing girders has been done with a wrench for ages, [literally even in this ancient fucking build,](http://i.imgur.com/1KjDHgC.jpg), and changing it would feel really inconsistent. Not to mention that now if you don't have welding protection and need to get through a quick wall you have to risk twice the eye damage for no real reason.
:+1: for the fix, big :-1: for the construction steps change.

kingofkosmos committed:
MADE FALSE WALL CONSTRUCTION TAKE SOME TIME (but only half the time of normal construction steps since it's not airtight and such).

Replaced numbered states with the GIRDER_stuff.

Fixed some grammar stuff.

Made reinforced girder deconstruction give the plasteel back when deconstructed.

kingofkosmos commented:
I hope I did the stuff RR said right... Tested and everything worked correctly.

I really hope to hear what people think about false walls being non-instant to build (like every other thing in the game), mostly hoping that it's ok to add that to this PR and that it's not such a big balance change that it requires a whole new PR?

phil235 commented on code/game/objects/structures/girders.dm:
```diff
@@ -1,51 +1,65 @@
+			playsound(src.loc, 'sound/items/Welder2.ogg', 50, 1)
+			user.visible_message("<span class='warning'>[user] disassembles the girder.</span>", \
+								"You start to disassemble the girder...", "You hear welding and clanking.")
+			var oldloc = src.loc
```
"var/oldloc" , not "var oldloc" please.

phil235 commented on code/game/objects/structures/girders.dm:
```diff
@@ -1,51 +1,65 @@
+			return
+		playsound(src.loc, 'sound/items/Ratchet.ogg', 100, 1)
+		user << "<span class='notice'>You start securing the girder...</span>"
+		var oldloc = src.loc
```
see above comment

phil235 commented on code/game/objects/structures/girders.dm:
```diff
@@ -95,23 +110,28 @@
 					if (S.use(2))
-						user << "<span class='notice'>You create a false wall! Push on it to open or close the passage.</span>"
+						user << "<span class='notice'>You start building a false wall...</span>"
+						var oldloc = src.loc
```
see above

phil235 commented on code/game/objects/structures/girders.dm:
```diff
@@ -123,22 +143,27 @@
 					S.use(2)
-					user << "<span class='notice'>You create a false wall! Push on it to open or close the passage.</span>"
+					user << "<span class='notice'>You start building a reinforced false wall...</span>"
+					var oldloc = src.loc
```
see above

kingofkosmos replied:
Check. Done to all of them.

kingofkosmos committed:
"var oldloc"s -> "var/oldloc"
Fixed the unsecured girder sliding below vents and scrubbers by placing it on layer 2.45.

phil235 commented:
> Not to mention that now if you don't have welding protection and need to get through a quick wall you have to risk twice the eye damage for no real reason.

Not really, you'd just use welder then wrench and move the displaced girder. No need to disassemble it if you only want to get through.

KorPhaeron commented:
Change its layer var.

Also why any delay at all? What does a 2 second delay add besides slight annoyance and tedium? Doesn't add any interesting gameplay, doesn't address a balance issue, just a delay for the sake of one

ACCount12 commented:
>You're not supposed to be able to break down walls with the emergency toolbox, it's only supposed to help with airlock opening (crowbaring and unwelding) imo.

I added miniwelder specially for disassembling a wall or two in case of emegrency.

kingofkosmos commented:
>Also why any delay at all? What does a 2 second delay add besides slight annoyance and tedium? Doesn't add any interesting gameplay, doesn't address a balance issue, just a delay for the sake of one

Mostly for the sake of consistency also since the majority of things take some time to build. Lightning-fast building feels out-of-place nowadays and makes no sense IC-wise either.

norill commented:
machine frames use wrench to disassemble, but computer frames use welder.

WJohn commented:
Machine frames should probably be anchorable with a wrench, and destroyed by a welder, then.

kingofkosmos commented:
Already working on that one and it just needs testing. Should I add that to this PR or make a new one for it?

phil235 commented:
Feel free to add it to that PR.

kingofkosmos committed:
dsfagregg

kingofkosmos committed:
Added machine frame disassembly with welder and securing/unsecuring with wrench.

kingofkosmos commented:
Added. Didn't know how to make it so that machine frames are built unsecured and also when a SMES (or any machine) is taken apart, it would create a secured machine frame instead of an unsecured one. So they are now already secured when built.
Tried to implement the new if(do_after(user, 40, target = src))-thing, but didn't get it to work, so it's done the old fashioned way.

Stuff has been tested and seems to be working fine.

Jordie0608 commented on code/game/machinery/constructable_frame.dm:
```diff
@@ -59,27 +59,67 @@
+					var oldloc = src.loc
 					if(do_after(user, 20))
+						if(src.loc!=oldloc)
+							return
```
do_after already handles this if you assign a target as one of the arguments.

Jordie0608 commented on code/game/machinery/constructable_frame.dm:
```diff
@@ -59,27 +59,67 @@
+						if( !WT.isOn() )
+							return
+						if(src.loc!=oldloc)
+							return
```
See above.

Jordie0608 commented on code/game/machinery/constructable_frame.dm:
```diff
@@ -59,27 +59,67 @@
+				var oldloc = src.loc
+				if(do_after(user, 40))
+					if(src.loc!=oldloc)
+						return
```
See above.

Jordie0608 commented on code/game/machinery/constructable_frame.dm:
```diff
@@ -59,27 +59,67 @@
+				var oldloc = src.loc
+				if(do_after(user, 40))
+					if(src.loc!=oldloc)
+						return
```
See above.

Jordie0608 commented on code/game/objects/structures/girders.dm:
```diff
@@ -1,51 +1,65 @@
-				var/obj/structure/girder/G = new (loc)
-				transfer_fingerprints_to(G)
+				if(src.loc!=oldloc)
+					return
```
See above.

Jordie0608 commented on code/game/objects/structures/girders.dm:
```diff
@@ -1,51 +1,65 @@
+		var/oldloc = src.loc
+		if(do_after(user, 40))
+			if(src.loc!=oldloc)
+				return
```
See above.

Jordie0608 commented on code/game/objects/structures/girders.dm:
```diff
@@ -95,23 +110,28 @@
+						var/oldloc = src.loc
+						if(do_after(user, 20))
+							if(src.loc!=oldloc)
+								return
```
See above.

Jordie0608 commented on code/game/objects/structures/girders.dm:
```diff
@@ -123,22 +143,27 @@
+					var/oldloc = src.loc
+					if(do_after(user, 20))
+						if(src.loc!=oldloc)
+							return
```
See above.

Jordie0608 commented on code/game/objects/structures/girders.dm:
```diff
@@ -271,24 +298,29 @@
+								"<span class='notice'>You start to disassemble the girder...</span>", "You hear welding and clanking.")
+			if(do_after(user, 40))
+				if( !WT.isOn() )
+					return
```
See above.

Jordie0608 commented:
Could you update the changelog for the machine frame changes.

kingofkosmos committed:
Update kingofkosmos-girderrakennus.yml

Added a mention about machine frames to the changelog.

kingofkosmos committed:
Do_afters now use "target = src", removed oldloc-stuff.
False wall construction had a bug if you had exactly 2 sheets in your hand, it wouldn't work - fixed it.

kingofkosmos committed:
Merge branch 'girderrakennus' of https://github.com/kingofkosmos/-tg-station into girderrakennus

kingofkosmos commented:
All done!

phil235 commented on code/game/machinery/constructable_frame.dm:
```diff
@@ -59,27 +59,55 @@
+										"<span class='notice'>You start to disassemble the frame...</span>", "You hear welding and clanking.")
+					if(do_after(user, 40, target = src))
+						if( !WT.isOn() )
+							return
```
Need a check here to make sure state is still equal to 1 after the do_after ends.

kingofkosmos replied:
So just to make this clear, like this?
					if(do_after(user, 40, target = src))
						if( !WT.isOn() )
							return
						if(state == 1)
							user << "<span class='notice'>You disassemble the frame.</span>"
							var/obj/item/stack/sheet/metal/M = new (loc, 5)
							M.add_fingerprint(user)
							qdel(src)

phil235 replied:
yes

phil235 replied:
You didn't add the check here.

kingofkosmos replied:
Whoops, it's there now.

phil235 commented on code/game/machinery/constructable_frame.dm:
```diff
@@ -59,27 +59,55 @@
-				user << "<span class='notice'>You dismantle the frame.</span>"
-				new /obj/item/stack/sheet/metal(src.loc, 5)
-				qdel(src)
+				if(do_after(user, 40, target = src))
```
same here, we don't want unanchored machine_frame with circuitboard and wire.

Supermichael777 commented:
feels weird that you need a welder to take apart something you didn't weld in the first place, but construction is weird. Also that picture of an old build MUH EYES

kingofkosmos committed:
if(state == 1) added to few places.

kingofkosmos commented:
Checks added.

kingofkosmos committed:
state check to disassembling the machine frame
